org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/CompilerConfiguration.java

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+* Copyright (c) 2024 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.core.compiler;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.internal.compiler.impl.CompilerOptions;
+
+/**
+ * This class encapsulates the standard compiler options that can be
+ * used to compile Java files. It provides methods to set and retrieve
+ * various compiler options, including source paths, class paths,
+ * output directories, annotation processing options, and other compiler
+ * arguments.
+ *
+ * Clients typically use this class when opting for an alternative compiler
+ * like Javac to compile Java files.
+ *
+ * @since 3.38
+ * @noextend This class is not intended to be subclassed by clients.
+ */
+public class CompilerConfiguration {
+	List<String> sourcepaths;
```

Comment:
It is not obvious that all those Strings refer to a file path. Better use a higher level Objects like org.eclipse.core.resources.IContainer for directory paths.

Reply:
I don't see it as a strong reason to use IContainer. I will keep using the file path.